### PR TITLE
Prevent MySQL connector from reading rows after close

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

File: presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcRecordCursor.java [hunk 1]
```diff
@@ -217,7 +218,7 @@ public void close()
             // do nothing
         }
         catch (SQLException e) {
-            throw new RuntimeException(e);
+            // ignore exception from close
```
Comment (findepi) [inline presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcRecordCursor.java]: Why is this needed?
Normally you do not want to suppress exceptions even in close. Maybe `MysqlClient`'s iml of `abortReadConnection` should swallow exc instead?

File: presto-mysql/src/main/java/com/facebook/presto/plugin/mysql/MySqlClient.java [hunk 2]
```diff
@@ -97,6 +98,15 @@ private static ConnectionFactory connectionFactory(BaseJdbcConfig config, MySqlC
         }
     }
 
+    @Override
+    public void abortReadConnection(Connection connection)
+            throws SQLException
+    {
+        // Abort connection before closing. Without this, the MySQL driver
+        // attempts to drain the connection by reading all the results.
```
Comment (findepi) [inline presto-mysql/src/main/java/com/facebook/presto/plugin/mysql/MySqlClient.java]: That's really cool feature ;)

File: presto-mysql/src/main/java/com/facebook/presto/plugin/mysql/MySqlClient.java [hunk 3]
```diff
@@ -97,6 +98,15 @@ private static ConnectionFactory connectionFactory(BaseJdbcConfig config, MySqlC
...
+        connection.abort(newDirectExecutorService());
```
Comment (findepi) [inline presto-mysql/src/main/java/com/facebook/presto/plugin/mysql/MySqlClient.java]: This should be `directExecutor()`

File: presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcRecordCursor.java (hunk 1 above)
Comment (dain) [reply presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcRecordCursor.java]: I agree normally you want to do something with this exception, but in this case MySQL throws a bogus (untyped) exception from close.  Also, in general, there is nothing interesting here for JDBC clients.

File: presto-mysql/src/main/java/com/facebook/presto/plugin/mysql/MySqlClient.java (hunk 2 above)
Comment (electrum) [reply presto-mysql/src/main/java/com/facebook/presto/plugin/mysql/MySqlClient.java]: A quote from our MySQL DBA:
> hm, you have some very long queries reading from DC-X into DC-Y, 50GB of data, is that expected?

File: presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcRecordCursor.java (hunk 1 above)
Comment (electrum) [reply presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcRecordCursor.java]: We discussed this and agreed there was no reason to spam the logs with stack traces from close exceptions. This is for a read, and at this point the data has already been read (or not due to an error), so can anything useful come from the exception?

Comment (findepi) [reply presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcRecordCursor.java]: In case of connection used once and only for a select query -- probably nothing useful. Unless it mean that a) a connection was not returned to a pool for some reason (when jdbc connector has pooling) or b) we're not terminating session correctly some times and leaking sessions on the db server (you know, db servers are sometimes stateful).

So -- i still see value in not swallowing exception here, but I am fine if this stays as-is.
